lib/typing/mongo/bson.go

```diff
@@ -103,12 +109,20 @@ func bsonValueToGoValue(value any) (any, error) {
 		return map[string]any{"$minKey": 1}, nil
 	case primitive.JavaScript:
 		return map[string]any{"$code": string(v)}, nil
+	case primitive.CodeWithScope:
+		return map[string]any{"$code": string(v.Code), "$scope": v.Scope}, nil
 	case
 		nil,
 		bool,
 		string,
 		int32, int64,
-		float32, float64:
+		float32:
+		return v, nil
+	case float64:
```

Comment:
Would we not want to check this for float32 too?

Reply:
Unless float32s can't be NaN/Inf?

Reply:
They theoretically can, but when you actually use the Infinity / -Infinity and NaN type, it creates float64

Reply:
BSON parses doubles into float64s

Reply:
To clarify, there should not be a way to get NaN / -/+ Inf that are float32 from BSON. But the language Go does technically allow it.